src/include/homeobject/shard_manager.hpp

```diff
@@ -27,11 +27,15 @@ struct ShardInfo {
     uint64_t deleted_capacity_bytes;
     std::optional< peer_id > current_leader{std::nullopt};
 };
+
 using InfoList = std::list< ShardInfo >;
+using ShardInfoPtr = std::shared_ptr< ShardInfo >;
+using ShardInfoPtrList = std::list< ShardInfoPtr >;
```

Comment:
It seems we do not need to expose these types above correct (move to `src/lib/homeobject_impl.hpp`)? We still return `std::list<ShardInfo>` in the Result<>.

Reply:
yes, these two types does not need to expose outside, will move to src/lib.